src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -693,6 +694,13 @@ string IDurableOrchestrationContext.StartNewOrchestration(string functionName, o
             }
             catch (TaskFailedException e)
             {
+                // Check to see if CallHttpAsync() threw a TimeoutException
+                // In this case, we want to throw a TimeoutException instead of a FunctionFailedException
+                if (functionName.Equals(HttpOptions.HttpTaskActivityReservedName) && e.InnerException is TimeoutException)
```

Comment:
Unrelated to this PR, but can you file an in depth issue of your findings here of why we needed this workaround, so we can investigate whether this impacts other exceptions customers will throw and if we can fix it in a non-breaking way?

Reply:
This was my suggestion to Varshi for how to know when to throw `TimeoutException` instead of `FunctionFailedException`. We're not really working around anything AFAIK. Surfacing outer exceptions other than `FunctionFailedException` is not something we expose to our customers.

Reply:
Makes sense.

I remember this being something we discussed around the V2 timeframe, but we missed our chance to tackle it before we locked in our breaking changes.

Probably worth keeping track of as a "breaking-change" issue that we can consider looking into if we make another breaking change release in the future (a ways off hopefully!).